packages/babel-core/src/transformation/file/options/option-manager.js

```diff
@@ -282,16 +282,27 @@ export default class OptionManager {
    */
   resolvePresets(presets: Array<string | Object>, dirname: string, onResolve?) {
     return presets.map(val => {
+      let options;
+      if (Array.isArray(val)){
+        [val, options] = val;
+      }
+
       if (typeof val === "string") {
         let presetLoc = resolve(`babel-preset-${val}`, dirname) || resolve(val, dirname);
         if (presetLoc) {
-          let val = require(presetLoc);
-          onResolve && onResolve(val, presetLoc);
-          return val;
+          val = require(presetLoc);
         } else {
           throw new Error(`Couldn't find preset ${JSON.stringify(val)} relative to directory ${JSON.stringify(dirname)}`);
         }
-      } else if (typeof val === "object") {
+      }
+
+      if (typeof val !== "function" && options !== undefined){
+        throw new Error(`Options ${JSON.stringify(options)} passed to a preset that does not accept options.`);
```

Comment:
Yeah, I went back and forth on it. I can show `presetLoc` if there is one, but showing `val` in general isn't that useful because plugins are just objects and the output ends up being not that helpful.